### DISCUSSION TITLE
Sunbird ED - Versioning Strategy, Which version is recommended for use ?

### DISCUSSION THREAD

**Discussion by parthlawate** [Sunbird-Ed/Community#328] [General Discussion]
Hello All, 

We have recently inherited an existing Sunbird ED installation to manage & enhance (This is a non-SAAS scenario. that is they are hosting their own Sunbird ED). They are running an older version of Sunbird.  
  
Generally while working with other open source projects, the policy we follow in recommending versions to use to customers is that they use the latest 'Long term Support' (LTS) release in their projects. In case we are inheriting existing projects, we ask that any new enhancements be done only after upgrading to the latest LTS. This will have several benefits as below : 

1. The customer gets the benefit of being on the latest LTS release which will be supported for some time to come by the community. 
2. This release will be more stable & have lot of bug fixes compared to the older release customer is using thus saving the customer cost of redoing bug fixes on their end
3. Any enhancements done will be more easily maintainable with a newer release & also have a shorter cycle to get merged in core if any contributions are being made. 

To sum it up, the larger the delta between the latest release & the release used by the customer, its more work & thus cost to maintain. Plus there is a higher risk of security vulnerabilities in older versions. 

Ofcourse upgrading to latest releases has to be simple enough to justify the benefits of using latest versions. Also its important to have some release strategy in place to make the decision making easier. 

While i've found the roadmaps & release notes for the 4x & 5x releases , I do not see any policy related to long term & short term releases. 

Also are we following Semantic versioning ? When should breaking changes be expected ? 

cc @Krishnaj20 @pramodkvarma

**Comment by Krishnaj20** [comment]
@parthlawate - Pls go thru the [existing](https://community.sunbird.org/) guidelines co-created with the Sunbird community mid of last year. You can contribute to this document by incorporating the best practices you have observed elsewhere, and are relevant in the Sunbird context. 
For this, 
a) I recommend you to go thru the existing document. Propose new changes to the existing process with the community folks tagged in this thread. 
b) Community can discuss and agree on what items can be incorporated. 
c) The agreed changes can then be adopted by the respective building blocks.

@surendrasinghs @chitranshu-keshav @vinukumar-vs @vrayulu @rhwarrier @vijiurs @pkranga @Prateek-slokam @aks30 @madhucr @alok-os

**Reply by parthlawate** [reply]
Thanks @Krishnaj20 .This answers my question about version naming & type of releases.  The other question i have is - as an adopter, what is the version recommended for use, What versions are said to be 'Unsupported' that aspect is not clear here.

**Reply by parthlawate** [reply]
@Krishnaj20 can we put out an official note about this on the community & link it here so we can close this topic ?

**Reply by Krishnaj20** [reply]
@parthlawate - we will recommend adopters be always on the latest version as this will be the only version supported by ED.

@chitranshu-keshav is preparing a doc which will be shared with us, DC, and few others who were involved in the initial discussion for feedback before publishing it to the community as an official note.

**Reply by parthlawate** [reply]
OK awesome thanks ! 

cc @coolbung @manojl
